### ISSUE TITLE
Namespacing using class leads to falsely undocumented classes

### ISSUE THREAD

**Issue by Overbryd** [inch-ci/inch_ci-web#111]
Inch CI says my class is undocumented, but it actually is. In the suggestion reference it used the class definition of one of the inner classes of this class.

See the fully documented class here:
https://github.com/Overbryd/adyen_client/blob/37fbd929a6693394b702fd69f96cc51fedbab86d/lib/adyen_client.rb

---
Re: [U] [AdyenClient](http://inch-ci.org/github/Overbryd/adyen_client/revision/37fbd929/code_object/1881953?branch=master) in [Overbryd/adyen_client](http://inch-ci.org/github/Overbryd/adyen_client)

**Comment by rrrene** [comment]
Hi there,

the class itself is actually undocumented. Class documentation precedes the class definition and often describes the intended purpose of the class and sometimes usage examples:

```ruby
# Base class for building linters that check source code
#
# Linters are used by authors of ROM adapters to verify that their
# integration complies with the ROM api.
#
# Most of the time, authors won't need to construct linters directly
# because the provided test helpers will automatically run when required
# in tests and specs.
#
# @example
#   require 'rom/lint/spec'
#
#
# @api public
class Linter
  # ... snip ...
end
```

[Example taken from ROM](https://github.com/rom-rb/rom/blob/a82fb42c3fdc5239e367a09904e7b8c517bdd63b/lib/rom/lint/linter.rb#L17)

**Comment by Overbryd** [comment]
Ah ok, good to know how inch sees this.
But it seems inch still sees the file lib/adyen_client/utils.rb as the main AdyenClient file.

https://github.com/inch-ci/inch_ci-web/issues/new?title=&body=%0A%0A---%0ARe:%20[U]%20[AdyenClient](http://inch-ci.org/github/Overbryd/adyen_client/revision/a4d6f53a/code_object/1881953?branch=master)%20in%20[Overbryd/adyen_client](http://inch-ci.org/github/Overbryd/adyen_client)

Although the class is now documented: https://github.com/Overbryd/adyen_client/blob/master/lib/adyen_client.rb

**Comment by rrrene** [comment]
You are right about the incorrect file detection. Will have to look into it!
